apps/src/JSInterpreter.js

```diff
@@ -559,14 +559,14 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
       }
       this.maxValidCallExpressionDepth = stackDepth;
 
-      if (inUserCode && this.interpreter.stateStack[0].node.type === "CallExpression") {
+      if (inUserCode && this.interpreter.stateStack[this.interpreter.stateStack.length - 1].node.type === "CallExpression") {
         // Store that we've seen a call expression at this depth in callExpressionSeenAtDepth:
         this.callExpressionSeenAtDepth[stackDepth] = true;
       }
 
       if (this.paused) {
         // Store the first call expression stack depth seen while in this step operation:
-        if (inUserCode && this.interpreter.stateStack[0].node.type === "CallExpression") {
+        if (inUserCode && this.interpreter.stateStack[this.interpreter.stateStack.length - 1].node.type === "CallExpression") {
```

Comment:
This and line 562 above would be a good place to extract `topFrame` - or even to extract the whole expression, something like `inUserCallExpression`.